lib/upgrader.ml

```diff
@@ -438,13 +438,22 @@ let old_doc = "old_doc"
 
 let name_converter name = [%string "$(convert)_$name"]
 
-let rec modification_to_string ?version_when_main_type = function
+let rec modification_to_string ~kind ?version_when_main_type =
+  let modification_to_string = modification_to_string ~kind in
+  function
   | Name (New type_name) ->
     [%string "fun _ -> $(make)_$type_name $converter $old_doc"]
   | Name (Modified type_name) ->
     [%string "$(name_converter type_name) $converter $old_doc"]
   | Option modification | Nullable modification ->
-    [%string "Option.map ($(modification_to_string modification))"]
+    let opt_map =
+      match kind with
+      | Config.Rescript ->
+        "Belt.Option.map"
+      | Config.Native ->
+        "Option.map"
+    in
+    [%string "$(opt_map) ($(modification_to_string modification))"]
```

Comment:
how can this work given `Belt.Option.map` is t-first while `Option.map` is t-last?

I think we should open Belt on top so every function uses Belt and t-first and use `|.` pipe-first instead of `|>` in code generation.